### PR TITLE
Warn if $ appears in an identifier

### PR DESCRIPTION
**Pull request by stonea**
This PR is for this issue: https://github.com/chapel-lang/chapel/issues/22979, which asks should we continue to permit `$` in identifier names?

This PR conducts a post-parse check to produce a warning on any declaration who's name has a $.

Take a look at the changes to see the impact. There are a few modules files that need to be updates and a handful of tests.

# TODO

* [x] Paratests
* [x] Paratests w/ gasnet

### CODE REVIEW

File: test/classes/hannah/configAtomicsNotSupported.chpl [hunk 1]
```diff
@@ -1 +1 @@
-config var n$ : atomic int;
+config var n : atomic int;
```
Comment (lydia-duncan) [inline test/classes/hannah/configAtomicsNotSupported.chpl:1]: Should there be a .compopts change with this?

Comment (stonea) [reply test/classes/hannah/configAtomicsNotSupported.chpl:1]: Ah, it's a config var good question.

I don't see a `.compopts` file for this test so I think not.

File: frontend/lib/uast/post-parse-checks.cpp [hunk 2]
```diff
@@ -1124,6 +1125,10 @@ void Visitor::checkReservedSymbolName(const NamedDecl* node) {
   } else if (isNameReservedType(name)) {
     error(node, "attempt to redefine reserved type '%s'.", name.c_str());
   }
+
+  if(strchr(name.c_str(), '$') != nullptr) {
+    warn(node, "Using $ in identifiers is deprecated; rename this to not use a $.");
```
Comment (arezaii) [inline frontend/lib/uast/post-parse-checks.cpp:1130]: I'm not sure on the style recommendation here, but I typically wrap things I'm calling out in the warnings/errors with single quotes. Here I would wrap the `$`s, but that may just be me.

**Review by arezaii — APPROVED** [review]
I would ask for a deprecation test to make sure the message gets emitted as expected, otherwise LGTM!

File: frontend/lib/uast/post-parse-checks.cpp (hunk 2 above)
Comment (stonea) [reply frontend/lib/uast/post-parse-checks.cpp]: There's no style guide for error message AFAIK, but I think quoting things like that makes sense.

I've updated it here: [`41566ff` (#23166)](https://github.com/chapel-lang/chapel/pull/23166/commits/41566ffe608f0b96ccc30f27bf590815826a6b1c)

**Pull request merged**